src/consumer.rs

```diff
@@ -137,6 +149,44 @@ impl Consumer {
     pub fn is_closed(&self) -> bool {
         self.internal.is_closed()
     }
+
+    pub async fn store_offset(&self, offset: u64) -> Result<(), ConsumerStoreOffsetError> {
+        if self.name.is_none() {
+            return Err(ConsumerStoreOffsetError::NameMissing);
+        }
+
+        let result = self
+            .internal
+            .client
+            .store_offset(
+                self.name.clone().unwrap().as_str(),
+                self.internal.stream.as_str(),
+                offset,
+            )
+            .await;
+
+        match result {
+            Ok(()) => Ok(()),
+            Err(e) => Err(ConsumerStoreOffsetError::Client(e)),
+        }
+    }
```

Comment:
You could make it a little bit shorter like this:

```suggestion
    pub async fn store_offset(&self, offset: u64) -> Result<(), ConsumerStoreOffsetError> {
        if let Some(name) = &self.name {
            self.internal
                .client
                .store_offset(name, self.internal.stream.as_str(), offset)
                .await
                .map(Ok)?
        } else {
            Err(ConsumerStoreOffsetError::NameMissing)
        }
    }
```

same for `query_offset`.